include/deal.II/lac/sparse_matrix.h

```diff
@@ -2377,6 +2384,35 @@ SparseMatrix<number>::end (const size_type r)
 }
 
 
+template <typename number>
+template <class StreamType>
+void SparseMatrix<number>::print_matrix_market(StreamType &out,
+					       const number threshold) const
+{
+  Assert(cols != 0, ExcNotInitialized());
+  Assert(val != 0, ExcNotInitialized());
+  Assert( threshold > 0, dealii::ExcMessage("Negative threshold!") ); 
+
+  //Print the header
+  out << "%%MatrixMarket matrix coordinate real general\n";
```

Comment:
maybe the easiest way is to make it work for complex as well. I would introduce two internal functions in anonymous namespace which would do the printing instead of
```
<< value
```
Something along the lines
```
template <class StreamType, typename number>
print_value (StreamType &out, const number &val)
```
and
```
template <class StreamType, typename number>
print_value (StreamType &out, const std::complex<number> &val)
```

You can probably generalise the header line (`real` or `complex`) the same way.

@bangerth would you agree or are there more straight forward ways to accomplish this?